rock-paper-scissors/service/src/main.rs

```diff
@@ -0,0 +1,218 @@
+use oasis_std::{Address, Context};
+#[macro_use]
+extern crate serde;
+
+#[derive(oasis_std::Service)]
+struct RockPaperScissors {
+    player_name: String,
+    player_move: Move, // TODO would be better as an enum?
+    challenger_name: String,
+    challenger_move: Move,
+    played: bool, // whether or not someone can challenge. true = player has played. false = nobody has played or just finished.
+    challenged: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub enum Move {
+    Null,
+    Rock,
+    Paper,
+    Scissors,
+}
+
+impl Move {
+    fn compare(&self, challenger: &Self) -> i32 {
```

Comment:
please use `PartialOrd`. also `x == y` is equivalent to `x.eq(&y)`

Reply:
PartialOrd requires transitive property, which doesn't apply to rock paper scissors